index.js

```diff
@@ -480,16 +480,31 @@ function multilineStyle(textarea: HTMLTextAreaElement, arg: StyleArgs) {
 
 function orderedList(textarea: HTMLTextAreaElement): SelectionRange {
   const orderedListRegex = /^\d+\.\s+/
+  const noInitialSelection = textarea.selectionStart === textarea.selectionEnd
   let selectionEnd
   let selectionStart
   let text = textarea.value.slice(textarea.selectionStart, textarea.selectionEnd)
+  let textToUnstyle = text
   let lines = text.split('\n')
-
-  const undoStyling = lines.every(line => orderedListRegex.test(line))
+  let startOfLine, endOfLine
+  if (noInitialSelection) {
+    const linesBefore = textarea.value.slice(0, textarea.selectionStart).split(/\n/)
+    startOfLine = textarea.selectionStart - linesBefore[linesBefore.length - 1].length
+    endOfLine = wordSelectionEnd(textarea.value, textarea.selectionStart, true)
+    textToUnstyle = textarea.value.slice(startOfLine, endOfLine)
+  }
+  const linesToUnstyle = textToUnstyle.split('\n')
+  const undoStyling = linesToUnstyle.every(line => orderedListRegex.test(line))
 
   if (undoStyling) {
-    lines = lines.map(line => line.replace(orderedListRegex, ''))
+    lines = linesToUnstyle.map(line => line.replace(orderedListRegex, ''))
     text = lines.join('\n')
+    if (noInitialSelection && startOfLine && endOfLine) {
+      const lengthDiff = linesToUnstyle[0].length - lines[0].length
+      selectionStart = selectionEnd = textarea.selectionStart - lengthDiff
+      textarea.selectionStart = startOfLine
+      textarea.selectionEnd = endOfLine
```

Comment:
Fix for 2.

`L504`: When there is no selection range, and styles were undone, set the cursor to the right position by the difference in length. For example:

```
25. fix an|other bug (index = at 10)
```

`'25. '.length === 4`

```
fix an|other bug (index = 10 - 4 = 6)
```

`L505-L506`: Change the textarea selection range so the current _line_ gets replaced with the new unstyled string.